lib/course_planner/offered_courses.ex

```diff
@@ -74,4 +76,29 @@ defmodule CoursePlanner.OfferedCourses do
      preload: [:teachers, :course, :term, classes: {c, attendances: a}],
      where: c.date < ^date and a.attendance_type == "Not filled")
   end
+
+  def create_pending_attendance_notification_map do
+      with_pending_attendances()
+      |> Enum.flat_map(fn(offered_course) ->
+        offered_course.teachers
+        |> Enum.map(fn(teacher) ->
+           %{
+              user: teacher,
+              type: :attendance_missing,
+              path: AttendanceHelper.get_offered_course_fill_attendance_path(offered_course.id)
+            }
+        end)
+      end)
+  end
+
+  def notify_missing_attendances do
+    create_pending_attendance_notification_map()
+    |> Enum.each(fn(email_data) ->
+         Notifications.new()
+         |> Notifications.type(email_data.type)
+         |> Notifications.resource_path(email_data.path)
+         |> Notifications.to(email_data.user)
+         |> @notifier.notify_later()
+       end)
+  end
```

Comment:
I'm not sure if this functions should belong in this module, maybe inside `UserEmail`?

Reply:
i merely followed the rest of the code

Reply:
Ah, my bad, it's doing the way you're doing. I guess I wanted to refactor it but it's not done yet.

Reply:
Although I kinda agree with you, i was already extracting this repetitive part back to notificatins